test/unixlistener/main.d

```diff
@@ -67,46 +141,105 @@ void writeToClient( cstring socket_path, cstring command )
 
 int main ( )
 {
-    auto local_address = "/tmp/ocean_unixsocket_test.socket";
-    auto epoll = new EpollSelectDispatcher();
-    auto unix_socket   = new UnixSocket;
-
-    // Value to be incremented via command to server
-    // Needs to be 3 after end of tests.
-    int expected_value = 0;
+    // Since this is a one-off test, we will not care about destroying
+    // these
+    start_semaphore = cast(sem_t*)mmap(null,
+            sem_t.sizeof, PROT_READ | PROT_WRITE,
+            MAP_SHARED | MAP_ANON, -1, 0);
 
-    void handleIncrementCommand ( cstring args,
-            void delegate ( cstring response ) send_response )
+    if (start_semaphore is null)
     {
-        expected_value += Integer.parse(args);
+        return 1;
     }
 
-    void handleShutdown ( cstring args,
-            void delegate ( cstring response ) send_response )
+    if (sem_init(start_semaphore, 1, 0) == -1)
     {
-        epoll.shutdown();
+        return 1;
     }
 
-    auto unix_server   = new UnixListener(local_address, epoll,
-            ["shutdown"[]: &handleShutdown,
-             "increment": &handleIncrementCommand]
-    );
-
-    epoll.register(unix_server);
-
-    // Write all to the socket, including shutdown, so we can just eventLoop
-    // and collect what's there
-
-    // Example passing arguments to handler:
-    writeToClient(local_address, "increment 2\n");
-    writeToClient(local_address, "increment 1\n");
-    writeToClient(local_address, "shutdown\n");
-
-    // Spin the server
-    epoll.eventLoop();
-
-    // This will be reached only if "shutdown" command was successful.
-    test!("==")(expected_value, 3);
+    auto local_address = "/tmp/ocean_unixsocket_test.socket";
```

Comment:
I suggest using a temporary directory created via `mkdtemp` just in case someone want's to run the same test concurrently. I know it sounds strange, but when building matrixes in automated builds it might happen.